flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/AccumulatorReport.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.taskexecutor;
+
+import org.apache.flink.runtime.accumulators.AccumulatorSnapshot;
+
+import java.io.Serializable;
+import java.util.Iterator;
+import java.util.List;
+
+/**
+ * A report about the current values of all accumulators of the TaskExecutor for a given job.
+ */
+public class AccumulatorReport implements Serializable, Iterable<AccumulatorSnapshot> {
```

Comment:
I think it would be better to not implement `Iterable`. `AccumulatorReport` is a value class which contains a `Collection` of `AccumulatorSnapshot`. Thus, it should be as simple as this. By letting this class implement the `Iterable` interface it basically means that `AccumulatorReport` can be used in places where an `Iterable` is usable. I think this should not be the case for this class. My concern is that we only let it implement this interface to cut a corner when iterating over the list of `AccumulatorSnapshots`.

Reply:
I can change it. I followed the design of `SlotReport` which also implements iterable even though it just holds a collection.

Reply:
I think `SlotReport` is wrongly implementing this interface.